OKlib.js

```diff
@@ -824,416 +767,378 @@ registerPlugin({
     }
 
     /**
-     * Concatenates two Arrays
-     * @param  {Object[]} array
-     * @param  {Object[]} elements
-     * @return {Object[]} Returns a new Array containing all Elements of the two given Arrays
+     * concatenates two arrays
+     * @param {Object[]} array
+     * @param {Object[]} elements
+     * @return {Object[]} returns a new array containing all elements of the two given arrays
      */
     function arrayCombineArrays(array, elements) {
         array = arrayCreateArray(array);
         elements = arrayCreateArray(elements);
         let result = [];
-        for (let i = 0; i < array.length; i++) {
+        for (let i in array) {
             result.push(array[i]);
         }
-        for (let j = 0; j < elements.length; j++) {
+        for (let j in elements) {
             result.push(elements[j]);
         }
-        log("arrayCombineArrays: Found '" + result.length + "' Objects", 4);
+        log('arrayCombineArrays: Found ' + result.length + ' Objects', 4);
         return result;
     }
 
     /**
-     * Checks if an Array contains the given Element
-     * @param  {Object[]} array
-     * @param  {Object} element
-     * @param  {Function} compare A Compare Function that should be used for the Comparison, if not set 'equal' is used
-     * @return {Boolean}  Returns true if the Element is contained in the Array
+     * checks if an array contains the given element
+     * @param {Object[]} array
+     * @param {Object} element
+     * @param {Function} compare a compare function that should be used for the comparison | if not set, 'equal' is used
+     * @return {Boolean} returns true if the element is contained in the array
      */
     function arrayContainsElement(array, element, compare) {
-        if (!compare) {
-            compare = equal;
-        }
-        for (let i = 0; i < array.length; i++) {
-            if (compare(array[i], element)) {
-                return true;
-            }
+        if (!compare) compare = equal;
+        for (let i in array) {
```

Comment:
change this to for..of or revert to normal for loop, because for..in is not safe for array operation (might not iterate in the right order)